config/parser.go

```diff
@@ -33,8 +36,37 @@ func ParseConfigFromYamlFile(path string, cfg interface{}) error {
 	if err != nil {
 		return err
 	}
+	return parseConfigFromYamlBytes(b, cfg)
+}
+
+func parseConfigFromYamlBytes(b []byte, cfg interface{}) error {
+	var data map[string]interface{}
+	err := yaml.Unmarshal(b, &data)
+	if err != nil {
+		return err
+	}
 
-	err = yaml.Unmarshal(b, cfg)
+	dc := &mapstructure.DecoderConfig{
+		DecodeHook: func(in, out reflect.Type, data interface{}) (interface{}, error) {
+			// Only intended to help with cases when string must be set to `time.Duration`
+			if in.Kind() != reflect.String || out.Name() != "Duration" {
```

Comment:
Checking by type name is not very certain. The [`reflect.Type`](https://golang.org/pkg/reflect/#Type) values are comparable:

```go
reflect.TypeOf(out) == reflect.TypeOf(time.Duration(0))
```
> [Play](https://play.golang.org/p/pI1btkA6uEO)